R/utils-time.R

```diff
@@ -13,18 +15,11 @@
 #' @importFrom lubridate hours
 #' @export
 flexibleTimeParse <- function(x, timezone, shiftTimeToNoon = TRUE) {
+  format_str <- c("Ymd HMOS z", "Ymd T* z*", "Ymd HMS")
```

Comment:
still unsure about this - if x is length 5, this will use `format_str[1]` to parse the first date, then `format_str[2]` to parse the second, then `format_str[3]` for the third date, then `format_str[1]` again for the 4th date, and `format_str[2]` for the fifth date. I don't think this is the behavior we want. We want it to test which format is most accurate.....

Perhaps something similar to this:
```r
library(lubridate)
timezone <- "UTC"
shiftTimeToNoon <- TRUE

# example has DV first (should be formatted in third statement)
# T* version second (should be formatted in second statement)
# third is still a wrong format so should remain NA in final time vector
x <- c("2010-10-01", "2017-01-01T12:12:13Z", "2017-01-01 12:12:55 UTC")

# first if statement becomes:
format_str <- "Ymd HMOS z"
time <- parse_date_time(x,format_str, tz=timezone,quiet = TRUE)
which.NA.1 <- sapply(time, repgen:::isEmptyOrBlank)

# second if statement becomes:
format_str <- "Ymd T* z*"
time[which.NA.1] <- parse_date_time(x[which.NA.1],format_str, tz=timezone,quiet = TRUE)
which.NA.2 <- sapply(time, repgen:::isEmptyOrBlank)

# third if statement becomes:
format_str <- "Ymd"
time[which.NA.2] <- parse_date_time(x[which.NA.2],format_str, tz=timezone,quiet = TRUE)
if (shiftTimeToNoon) {
  time[which.NA.2] <- time[which.NA.2] + hours(12)
}

time
```